WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVM.kt

```diff
@@ -97,6 +121,24 @@ class SiteCreationMainVM @Inject constructor(
         }
     }
 
+    fun preloadThumbnails(context: Context) {
+        if (preloadingJob == null) {
+            preloadingJob = viewModelScope.launch(Dispatchers.IO) {
+                if (networkUtils.isNetworkAvailable()) {
+                    val response = fetchHomePageLayoutsUseCase.fetchStarterDesigns()
+                    for (design in response.designs) {
+                        Glide.with(context)
+                                .downloadOnly()
+                                .load(design.previewMobile)
+                                .submit()
+                                .get() // This makes each call blocking, so subsequent calls can be cancelled if needed.
+                    }
+                    preloadingJob = null
```

Comment:
We could move the null assignment outside the network check block

Reply:
Good catch! Fixed with https://github.com/wordpress-mobile/WordPress-Android/pull/16584/commits/290593e06f786d7bb35a50c8639eefa51c4c4020 :+1: